### ISSUE TITLE
Training ControlNet-Xs 

### ISSUE THREAD

**Issue by nguyendung1612** [huggingface/diffusers#8382]
Thank you @UmerHA for contributing the ControlNet demo. I tried to create a training script so I could train the model on my data but had some problems using the UNetControlNetXSModel. 
Can you provide training or give me some suggestions?

**Comment by UmerHA** [comment]
What were the problems you ran into?

**Comment by nguyendung1612** [comment]
> What were the problems you ran into?

I tried 2 ways:

1) I trained the UNetControlNetXSModel and saved the pretrained model. But when I checked your Pipeline and yours pretrain, I saw that you only saved the ControlNetXSAdapter class. How can you save ControlNetXSAdapter while training UNetControlNetXSModel?

2) I trained ControlNetXSAdapter class (optimizer and scheduler for this), but my loss did not change during training. I think it's because I only used UNetControlNetXSModel for forward without optimizing it.

**Comment by sayakpaul** [comment]
It should be turned into a discussion.
